pkg/controller/controller.go

```diff
@@ -34,12 +34,12 @@ type AppGwIngressController struct {
 // NewAppGwIngressController constructs a controller object.
 func NewAppGwIngressController(kubeclient kubernetes.Interface, appGwClient network.ApplicationGatewaysClient, appGwIdentifier appgw.Identifier, k8sContext *k8scontext.Context) *AppGwIngressController {
 	controller := &AppGwIngressController{
+		appGwClient: appGwClient,
 		appGwIdentifier:  appGwIdentifier,
 		k8sContext:       k8sContext,
 		k8sUpdateChannel: k8sContext.UpdateChannel,
 	}
 	controller.eventQueue = NewEventQueue(controller.processEvent)
```

Comment:
We could also move eventQueue initialization up.

Reply:
@vramakrishnan i think that is not true. https://play.golang.org/p/-VciWJJPDgz

Reply:
You are right. Callback needs to be initialized in next step, the way it is today.